galsim/roman/roman_psfs.py

```diff
@@ -166,94 +166,136 @@ def getPSF(SCA, bandpass,
-        else:
-            if bandpass not in default_bandpass_list+['short','long'] and bandpass is not None:
-                raise GalSimValueError("Bandpass not a valid Roman bandpass or 'short'/'long'.",
-                                       bandpass, default_bandpass_list)
+        raise GalSimValueError("Bandpass not a valid Roman bandpass or 'short'/'long'.",
+                               bandpass, default_bandpass_list)
 
     # If bandpass is 'short'/'long', then make sure that interpolation is not called for, since that
     # requires an actual bandpass.
     if bandpass in ['short','long'] and n_waves is not None:
         raise GalSimValueError("Cannot use bandpass='short'/'long' with interpolation.", bandpass)
 
+    if not isinstance(wavelength, (Bandpass, float, type(None))):
+        raise TypeError("wavelength should either be a Bandpass, float, or None.")
+
     # Now call _get_single_PSF().
-    psf = _get_single_PSF(SCA, bandpass, SCA_pos, approximate_struts,
-                          n_waves, extra_aberrations, logger, wavelength,
-                          high_accuracy, pupil_plane_type, gsparams)
+    psf = _get_single_PSF(SCA, bandpass, SCA_pos, pupil_bin,
+                          n_waves, extra_aberrations, wavelength,
+                          pupil_plane_type, gsparams)
     return psf
 
-def _get_single_PSF(SCA, bandpass, SCA_pos, approximate_struts,
-                    n_waves, extra_aberrations, logger, wavelength,
-                    high_accuracy, pupil_plane_type, gsparams):
+def __make_aperture(SCA, pupil_plane_type, pupil_bin, wave, gsparams):
+    from . import diameter, obscuration
+    from .. import fits
+    from .. import meta_data
+    from ..phase_psf import Aperture
+
+    # Load the pupil plane image.
+    if pupil_plane_type == 'long':
+        pupil_plane_im = os.path.join(meta_data.share_dir, 'roman',
+            'SCA%d'%SCA + pupil_plane_file_longwave)
+    else:
+        pupil_plane_im = os.path.join(meta_data.share_dir, 'roman',
+            'SCA%d'%SCA + pupil_plane_file_shortwave)
+    pupil_plane_im = fits.read(pupil_plane_im, read_header=True)
+    # Native pixel scale in the file is for the exit pupil.  We want the scale of the
+    # entrance pupil.  Fortunately, they provide the conversion as PUPILMAG in the header.
+    # They also use microns for units, and we want meters, hence the extra 1.e-6.
+    pupil_plane_im.scale *= pupil_plane_im.header['PUPILMAG'] * 1.e-6
+
+    pupil_plane_im = pupil_plane_im.bin(pupil_bin,pupil_bin)
+
+    aper = Aperture(lam=wave, diam=diameter,
+                    obscuration=obscuration,
+                    pupil_plane_im=pupil_plane_im,
+                    gsparams=gsparams)
+    return aper
+
+# Usually a given run will only need one or a few different apertures for repeated getPSF calls.
+# So cache those apertures here to avoid having to remake them.
+_make_aperture = LRU_Cache(__make_aperture)
```

Comment:
Is it worth mentioning this caching in the docstring for getPSF() so that users are aware that the potentially significant setup costs may not apply for each getPSF() call?